### PR TITLE
ui: worked on DQ feedback part 3

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
Worked on DQ feedback
- integrated permission for test suite and test case
- added test case re-indexing support
- updated dynamic name of test suite ingestion pipeline
- added support to updated displayName of test case
- updated UI of "Add Test Case form" same as add test case modal of logical test suite
- other miner feedbacks

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by karanh37 — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/DataQuality/SummaryPannel/SummaryPanel.component.tsx [hunk 1]
```diff
@@ -40,8 +43,10 @@ export const SummaryPanel = ({ testSuiteId }: SummaryPanelProps) => {
   };
 
   useEffect(() => {
-    fetchTestSummary();
-  }, [testSuiteId]);
+    if (testCasePermission?.ViewAll || testCasePermission?.ViewBasic) {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataQuality/SummaryPannel/SummaryPanel.component.tsx:46]: Isn't it should be `ViewTests`?
Screenshot 2023-06-26 at 9 39 03 PM

File: openmetadata-ui/src/main/resources/ui/src/components/DataQuality/TestCases/TestCases.component.tsx [hunk 2]
```diff
@@ -151,14 +173,16 @@ export const TestCases = () => {
   };
 
   useEffect(() => {
-    if (tab === DataQualityPageTabs.TEST_CASES) {
-      if (searchValue) {
-        searchTestCases();
-      } else {
-        fetchTestCases();
+    if (testCasePermission?.ViewAll || testCasePermission?.ViewBasic) {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataQuality/TestCases/TestCases.component.tsx:176]: same viewTests

File: openmetadata-ui/src/main/resources/ui/src/components/DataQuality/TestSuites/TestSuites.component.tsx [hunk 3]
```diff
@@ -177,8 +180,10 @@ export const TestSuites = () => {
   };
 
   useEffect(() => {
-    fetchTestSuites();
-  }, [tab]);
+    if (testSuitePermission?.ViewAll || testSuitePermission?.ViewBasic) {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataQuality/TestSuites/TestSuites.component.tsx:183]: same

Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/DataQuality/TestSuites/TestSuites.component.tsx:183]: as this is testSuite list, and not any particular detail view, i am checking permission from resource.
image

File: openmetadata-ui/src/main/resources/ui/src/components/DataQuality/TestCases/TestCases.component.tsx (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/DataQuality/TestCases/TestCases.component.tsx:176]: Same here checking from resource permission

File: openmetadata-ui/src/main/resources/ui/src/components/DataQuality/SummaryPannel/SummaryPanel.component.tsx (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/DataQuality/SummaryPannel/SummaryPanel.component.tsx:46]: Same here checking from resource permission

**Pull request merged**